src/main/java/org/jabref/gui/preferences/NetworkTabViewModel.java

```diff
@@ -215,6 +223,56 @@ public boolean validateSettings() {
         return true;
     }
 
+    /**
+     * Check the connection by using the given url. Used for validating the http proxy.
+     * The checking result will be appear when request finished.
+     * The checking result could be either success or fail, if fail, the cause will be displayed.
+     */
+    public void checkConnection() {
+        DialogPane dialogPane = new DialogPane();
+        GridPane settingsPane = new GridPane();
+        settingsPane.setHgap(4.0);
+        settingsPane.setVgap(4.0);
+
+        Label messageLabel = new Label();
+        messageLabel.setText(Localization.lang("Warning: your settings will be saved.\nEnter any URL to check connection to:"));
+
+        TextField url = new TextField();
+        url.setText("http://");
+
+        settingsPane.add(messageLabel, 1, 0);
+
+        settingsPane.add(url, 1, 1);
+        dialogPane.setContent(settingsPane);
+        String title = Localization.lang("Check Proxy Setting");
+        dialogService.showCustomDialogAndWait(
+                title,
+                dialogPane,
+                ButtonType.OK, ButtonType.CANCEL)
+                .ifPresent(btn -> {
+                            if (btn == ButtonType.OK) {
+                                String connectionProblemText = Localization.lang("Problem with connection: ");
+                                String connectionSuccessText = Localization.lang("Connection Successful!");
```

Comment:
```suggestion
                                String connectionSuccessText = Localization.lang("Connection successful!");
```

No capitalization in text.